components/hab/src/command/studio/enter.rs

```diff
@@ -240,13 +240,20 @@ mod inner {
             return false;
         }
 
+        for arg in args.iter() {
+            let str_arg = arg.to_string_lossy();
+            if str_arg == String::from("-D") {
+                return false;
+            }
+        }
+
```

Comment:
Since this is nearly identical to the next block, I think it could be better simplified to:
```rust
        for arg in args.iter() {
            match arg.to_str().unwrap_or("") {
                "-D" => return false,
                "-w" | "--windows" => return true,
                _ => (),
            }
        }
```
This also avoids the need for as many objects to be created by using `&str`s instead of `String`s.

There is one small change here, the conversion to lower case is removed, but I think that's actually incorrect; `-W` (currently unused) shouldn't necessarily be treated the same as `-w`. Also, I don't see anywhere that `--windows` is a documented option, so I think we should either remove that or document it.

Reply:
`--windows` comes from `hab pkg build` which forwards to here. I had thought about doing what you suggest above but my concern was if someone included both flags. I know thats an edge case and technically erroneous but the `-D` flag should trump.